test/stripe/api_operations_test.rb

```diff
@@ -89,6 +89,36 @@ def self.object_name
         assert_equal "bar", nested_resource.foo
       end
 
+      should "define a retrieve method with only opts" do
+        stub_request(:get, "#{Stripe.api_base}/v1/mainresources/id/nesteds/nested_id")
+          .to_return(body: JSON.generate(id: "nested_id", object: "nested", foo: "bar"))
+        nested_resource = MainResource.retrieve_nested("id", "nested_id", { stripe_account: "acct_123" })
+        assert_equal "bar", nested_resource.foo
+      end
+
+      should "define a retrieve method with both opts and params" do
+        stub_request(:get, "#{Stripe.api_base}/v1/mainresources/id/nesteds/nested_id?expand[]=reverse")
+          .to_return(body: JSON.generate(id: "nested_id", object: "nested", foo: "bar"))
+        nested_resource = MainResource.retrieve_nested("id", "nested_id", { expand: ["reverse"] }, { stripe_account: "acct_123" })
```

Comment:
one last tiny nit: We could also check that the `Stripe-Account` header is being send here. I see that's being tested already in `TransferTest` though so I'm fine either way